mempool/v0/reactor_test.go

```diff
@@ -137,6 +148,11 @@ func TestReactorNoBroadcastToSender(t *testing.T) {
 			if err := r.Stop(); err != nil {
 				assert.NoError(t, err)
 			}
+
+			sw := r.Switch
+			for _, peer := range sw.Peers().List() {
+				sw.StopPeerGracefully(peer)
+			}
```

Comment:
Instead of doing this for every test, can we instead have this within the `makeAndConnectReactors` function i.e. can you pass in `t` and then run something like
```go
func makeAndConnectReactors(config *cfg.Config, n int) []*Reactor {
...
switches := p2p.MakeConnectedSwitches(config.P2P, n, func(i int, s *p2p.Switch) *p2p.Switch {
	s.AddReactor("MEMPOOL", reactors[i])
	return s

}, p2p.Connect2Switches)

t.Cleanup(func() {
    for _, switch := range switches {
        switch.Stop()
    }
})
return reactors
```
---
It would also be great if you could apply this to the `evidence` package.

Reply:
I didn't do `switch.Stop()` because the switch also stops the reactor, if we do call it, it will return an error because the reactor has been stopped before the Cleanup, but we can ignore the error for sure. Will update it in `evidence` package

Reply:
looks like the `leaktest` is pointing out the other goroutine error if run `go test -run TestBroadcastTxForPeerStopsWhenReactorStops -count 1000`

Reply:
> I didn't do switch.Stop() because the switch also stops the reactor

Well if it works out then perhaps you don't even need to worry about stopping the reactors then 😄